ssl-config/src/main/java/com/palantir/remoting/ssl/TrustStoreConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.ssl;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@JsonDeserialize(as = ImmutableTrustStoreConfiguration.class)
+@JsonSerialize(as = ImmutableTrustStoreConfiguration.class)
+@Value.Immutable
+@Value.Style(visibility = ImplementationVisibility.PACKAGE)
+public abstract class TrustStoreConfiguration {
```

Comment:
Will defer to library owners on that one -- just figured that it wouldn't be very polite for an external PR to force a major break/version change :).

If you/Mark are okay with this being a breaking change, happy to remove the deprecated files (although at this point maybe we can just do that as a follow-up commit so that I can shepherd this PR through -- would prefer to wrap this one up soon)